Sources/PackageGraph/PinsStore.swift

```diff
@@ -152,43 +159,52 @@ private struct PinsStorage {
         if !self.fileSystem.exists(self.path) {
-            return [:]
+            return (pins: [:], originHash: .none)
         }
 
         return try self.fileSystem.withLock(on: self.lockFilePath, type: .shared) {
             let version = try self.decoder.decode(path: self.path, fileSystem: self.fileSystem, as: Version.self)
             switch version.version {
             case V1.version:
                 let v1 = try decoder.decode(path: self.path, fileSystem: self.fileSystem, as: V1.self)
-                return try v1.object.pins.map { try PinsStore.Pin($0, mirrors: mirrors) }
-                    .reduce(into: [PackageIdentity: PinsStore.Pin]()) { partial, iterator in
-                        if partial.keys.contains(iterator.packageRef.identity) {
-                            throw StringError("duplicated entry for package \"\(iterator.packageRef.identity)\"")
-                        }
-                        partial[iterator.packageRef.identity] = iterator
-                    }
+                return (
+                    pins: try v1.object.pins
+                        .map { try PinsStore.Pin($0, mirrors: mirrors) }
+                        .reduce(into: [PackageIdentity: PinsStore.Pin]()) { partial, iterator in
+                            if partial.keys.contains(iterator.packageRef.identity) {
+                                throw StringError("duplicated entry for package \"\(iterator.packageRef.identity)\"")
+                            }
+                            partial[iterator.packageRef.identity] = iterator
+                        },
+                    originHash: .none
+                )
             case V2.version:
                 let v2 = try decoder.decode(path: self.path, fileSystem: self.fileSystem, as: V2.self)
-                return try v2.pins.map { try PinsStore.Pin($0, mirrors: mirrors) }
-                    .reduce(into: [PackageIdentity: PinsStore.Pin]()) { partial, iterator in
-                        if partial.keys.contains(iterator.packageRef.identity) {
-                            throw StringError("duplicated entry for package \"\(iterator.packageRef.identity)\"")
-                        }
-                        partial[iterator.packageRef.identity] = iterator
-                    }
+                return (
+                    pins: try v2.pins
+                        .map { try PinsStore.Pin($0, mirrors: mirrors) }
+                        .reduce(into: [PackageIdentity: PinsStore.Pin]()) { partial, iterator in
+                            if partial.keys.contains(iterator.packageRef.identity) {
+                                throw StringError("duplicated entry for package \"\(iterator.packageRef.identity)\"")
+                            }
+                            partial[iterator.packageRef.identity] = iterator
+                        },
+                    originHash: v2.originHash
```

Comment:
Should `originHash` be a thing only in a version 3? Basically, older SwiftPMs will remove the hash when resolving, so we might be getting confusing bug reports that people are still getting the old behavior, so might be good to just make it explicit.

Reply:
the tradeoff is that if we do that, old SwiftPM will error out on unknown resolved file version which may be annoying?

Reply:
yea we should be able too that, which means you only get the new resolved file behavior on packages that adopt the new tools version, but maybe that is okay?

Reply:
Yah, it just seems cleaner to me. 

Especially since it feels like users are (understandably!) confused by the current behavior and if switching between SwiftPM versions will continue to bring that confusion, we wouldn't solve that part of the issue.

Reply:
done!